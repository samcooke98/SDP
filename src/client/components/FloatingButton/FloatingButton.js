//You could call this file {ComponentName}.js as well, but you would have to ensure your imports are correct

import React from "react";
import style from "./web.css"

export default class FloatingButton extends React.Component {
    render() {
        return (
            <div 
                className={ style[this.props.shape] + " " + style.floatBtn}
                style= {
                    {
                        cursor: "pointer",
                        backgroundColor: this.props.colour,
                        color: this.props.color,
                        bottom: this.props.bottom,
                        right: this.props.right,
                        width: this.props.width,
                        height: this.props.height
                    }
                }
                onClick={this.props.onClick}
            >
            <span style={{textAlign: "center", verticalAlign: "middle", lineHeight: "55px",fontSize: "40px"}}>
                {this.props.children}
            </span>
            </div>
        )

    }

    someFunc = () => {
        //'This' is automatically bound when writing functions like this
        return JSON.stringify(this.props);
    }
}

FloatingButton.defaultProps = { 
    colour: "#2F80ED",
    shape: 'round'
}

//In addition to CSS modules, you may wish to use: https://github.com/gajus/react-css-modules