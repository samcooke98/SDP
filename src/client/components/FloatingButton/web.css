.floatBtn {
    position: fixed;
    bottom: 42px;
    right: 38px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, .25);
    display:flex;
    justify-content:center;
    align-items: center;
}

.round {
    width: 80px;
    height: 80px;
    border-radius: 40px;
}

.square { 
    width: 80px;
    height: 80px;   
    border-radius: 22.67px;
}

.floatBtn > * { 
    width: 64px;
    height:64px;

    fill:white;
    color: white;
}