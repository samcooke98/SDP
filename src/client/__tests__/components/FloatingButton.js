import React from "react";
import renderer from "react-test-renderer";
// import ReactDOM from 'react-dom';

import FloatingButton from "../../components/FloatingButton/FloatingButton";


describe('React Component: FloatingButton', () => {
    it("matches a snapshot", () => {
        const tree = renderer.create(
            <FloatingButton />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})