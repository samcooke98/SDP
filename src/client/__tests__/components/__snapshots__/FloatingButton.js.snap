// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component: FloatingButton matches a snapshot 1`] = `
<div
  className="undefined undefined "
  onClick={undefined}
  style={
    Object {
      "backgroundColor": "#2F80ED",
      "bottom": undefined,
      "color": undefined,
      "cursor": "pointer",
      "height": undefined,
      "right": undefined,
      "width": undefined,
    }
  }
>
  <span
    style={
      Object {
        "fontSize": "40px",
        "lineHeight": "55px",
        "textAlign": "center",
        "verticalAlign": "middle",
      }
    }
  />
</div>
`;
